src/Simplify.elm

```diff
@@ -10115,61 +10105,73 @@ Examples:
   - `String.concat (List.repeat n x) --> String.repeat n x`
   - `Animation.loop (List.repeat n x) --> Animation.repeat n x` using [`mdgriffith/elm-style-animation`](https://package.elm-lang.org/packages/mdgriffith/elm-style-animation/4.0.0/)
   - `FormattedText.concat (List.intersperse s list) --> FormattedText.join s list` using [`NoRedInk/elm-formatted-text-19`](https://package.elm-lang.org/packages/NoRedInk/elm-formatted-text-19/1.0.0/)
-
--}
-callFromCanBeCombinedCheck :
-    { fromFn : ( ModuleName, String ), combinedFn : ( ModuleName, String ) }
-    -> IntoFnCheck
-callFromCanBeCombinedCheck config =
-    onFnCallCanBeCombinedCheck
-        { laterOperationDescription = \fn -> qualifiedToString fn
-        , earlierFn = config.fromFn
-        , combinedFn = config.combinedFn
-        }
-
-
-{-| Simplify this operation after a given call to `earlierFn` into a given `combinedFn`.
-
-Examples:
-
   - `List.filterMap identity (List.map f list) --> List.filterMap f list`
   - `List.filterMap identity << List.map f --> List.filterMap f`
   - `traverse identity (map f a) --> traverse f a`
-  - those listed in `callFromCanBeCombinedCheck`
+  - those listed in `onSpecificFnCallCanBeCombinedCheck`
 
 -}
-onFnCallCanBeCombinedCheck :
-    { laterOperationDescription : ( ModuleName, String ) -> String
+onSpecificFnCallCanBeCombinedCheck :
+    { args : List { argProperties | description : String, is : Infer.Resources {} -> Node Expression -> Bool }
     , earlierFn : ( ModuleName, String )
     , combinedFn : ( ModuleName, String )
     }
     -> IntoFnCheck
-onFnCallCanBeCombinedCheck config =
+onSpecificFnCallCanBeCombinedCheck config =
+    let
+        laterOperationDescription : ( ModuleName, String ) -> String
+        laterOperationDescription laterFn =
+            case config.args of
+                [] ->
+                    qualifiedToString (qualify laterFn defaultQualifyResources)
+
+                firstArg :: argsAfterFirst ->
+                    qualifiedToString (qualify laterFn defaultQualifyResources)
+                        ++ " with "
+                        ++ firstArg.description
+                        ++ String.concat (List.map (\arg -> " and " ++ arg.description) argsAfterFirst)
+
+        laterInitArgsAreValidForOperation : List (Node Expression) -> Infer.Resources res -> Bool
+        laterInitArgsAreValidForOperation laterInitArgs resources =
+            List.all identity
+                (List.map2 (\arg argConfig -> argConfig.is (extractInferResources resources) arg)
+                    laterInitArgs
+                    config.args
```

Comment:
Could there be a problem here if the length of the 2 lists is not the same? :thinking:

Reply:
That's why we additionally check for `fullyAppliedLastArg` and `onlyLastArgIsCurried checkInfo.later` in the call and composition cases.